src/pages/chainlink-functions/resources/add-functions-to-projects.md

```diff
@@ -0,0 +1,317 @@
+
+1. Save the file and run the script:
+
+   ```shell
+   npx hardhat run scripts/functions-sub.js
+   ```
+
+1. If the script is successful, you the terminal prints your subscription ID. Record this ID to use for Chainlink Functions requests. You should see output similar to the following example:
+
+   ```text
+   Subscription created with ID: 1337
+   Duplicate definition of Transfer (Transfer(address,address,uint256,bytes), Transfer(address,address,uint256))
+   Funding with 1000000000000000000 LINK
+   Subscription 1337 funded with 1000000000000000000 LINK
+   Adding consumer contract address 0xE83188aDF26bd6c18076704fC8EC68CBda454f76 to subscription 1337
+   Authorized consumer contract: 0xE83188aDF26bd6c18076704fC8EC68CBda454f76
+   ```
+
+Now that the consumer contract is deployed and the subscription is created and funded with LINK, the on-chain resources are ready to handle your requests. Next, create JavaScript code that you want to run on the DON, configure arguments for the code, and create a script to send your request.
+
+## Send requests
+
+After your on-chain resources are configured, you can send Chainlink Functions requests to the DON. This can be done from a Web3 application, script, another on-chain smart contract, or any other location capable of submitting requests to your consumer contract. This example uses [Ethers v6](https://docs.ethers.org/v6/) without Hardhat to show you how to generate a request, encrypt secrets, send your request, and read the fulfillment response.
+
+Each request has the following components:
+
+- **Source code:** JavaScript code that will run on the DON.
+- **Arguments:** Optional arguments for the source code. The arguments that you need are defined in your source code. Depending on how you configured your source, you might not need any arguments at all.
+- **Secrets:** Optional secrets that your source code needs to access APIs or other interfaces. Secrets can be included inline with your request or hosted off-chain. See the [Using Secrets in Requests](/chainlink-functions/tutorials/api-use-secrets) and [Using Off-chain Secrets](/chainlink-functions/tutorials/api-use-secrets-offchain) tutorials for examples.
+- **Secrets location:** This tells the DON where to obtain your secrets.
+
+### Create a request script
+
+Download some example source code, and create a script to assemble the required components. The script will read your source code, define arguments, encrypt secrets, and send requests to your consumer contract. This example script does not require Hardhat, so you can modify it to run in a browser using Ethers or another framework.
+
+1. Use `curl` to get the example source code. This code runs on each node in the DON and returns a response to your consumer contract. For this example, use the source from the [Call an API](/chainlink-functions/tutorials/api-query-parameters) tutorial. The following `curl` request creates a file named `Functions-request-source.js` with the source code:
```

Comment:
do we need to specify here how this is showing an example, but if they have their own JS code, they should just skip this and refernece it in the next steps?

Reply:
Good suggestion. I clarified that this is the point where you can include your own source file. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0